api/api_ethereum.go

```diff
@@ -917,3 +1166,164 @@ func (api *EthereumAPI) rpcMarshalHeader(head *types.Header) (map[string]interfa
 
 	return result, nil
 }
+
+func EthDoCall(ctx context.Context, b Backend, args EthTransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *EthStateOverride, timeout time.Duration, globalGasCap uint64) ([]byte, uint64, error) {
+	defer func(start time.Time) { logger.Debug("Executing EVM call finished", "runtime", time.Since(start)) }(time.Now())
+
+	st, header, err := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if st == nil || err != nil {
+		return nil, 0, err
+	}
+	if err := overrides.Apply(st); err != nil {
+		return nil, 0, err
+	}
+
+	// Setup context so it may be cancelled the call has completed
+	// or, in case of unmetered gas, setup a context with a timeout.
+	var cancel context.CancelFunc
+	if timeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+	// Make sure the context is cancelled when the call has completed
+	// this makes sure resources are cleaned up.
+	defer cancel()
+
+	// TODO-Klaytn: Klaytn is using fixed baseFee as now.
```

Comment:
Is this correct todo comment? I don't get it what we need to do from this.

Reply:
Oh, I will add some context about it.
What I wanted to say is: "Klaytn is using fixed baseFee as now. If we change this fixed baseFee as dynamic baseFee, we should update this logic too."

Reply:
Added some context [in here](https://github.com/klaytn/klaytn/pull/1134/commits/15cb596fffac22f51e4841d7f16f9dbe981d5b54)